src/Servers/Kestrel/Transport.Quic/src/WebHostBuilderQuicExtensions.cs

```diff
@@ -20,15 +19,11 @@ public static class WebHostBuilderQuicExtensions
     /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
     public static IWebHostBuilder UseQuic(this IWebHostBuilder hostBuilder)
     {
-        if (QuicListener.IsSupported)
+        return hostBuilder.ConfigureServices(services =>
```

Comment:
As @JamesNK mentioned [here](https://github.com/dotnet/aspnetcore/pull/44755/files#r1006327156), this breaks all the checks that assume the presence of a multiplexed factory indicates QUIC support and a transport that can potentially bind (e.g. [here](https://github.com/dotnet/aspnetcore/pull/47209/files#diff-0041b55fab5f441344d431c2c49f6770b3481df02d634b7feedb646c2f0c9281R51))

Reply:
That's why we're going to use check `IConnectionListenerFactorySelector.CanBind` rather than the presence of the `IMultiplexedConnectionListenerFactory` to determine whether QUIC is supported where necessary, right?

Reply:
How?  Wouldn't you need to pass in an endpoint?

Reply:
The check is here, right? With a similar check just below it in the same method.

https://github.com/dotnet/aspnetcore/blob/7aca30363667b13f80aae816dc9924de54284569/src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs#L184-L187

So `options.EndPoint`, I'd assume. If none of the `_multiplexedTransportFactories` both `IConnectionListenerFactorySelector` and say they can bind to the endpoint, and HTTP/3 isn't going to work.